proc-macros/src/render_client.rs

```diff
@@ -72,19 +72,15 @@ impl RpcDescription {
 	/// Verify and rewrite the return type (for methods).
 	fn return_result_type(&self, mut ty: syn::Type) -> TokenStream2 {
 		// We expect a valid type path.
-		let syn::Type::Path(ref mut type_path) = ty else {
-			return quote_spanned!(ty.span() => compile_error!("Expecting something like 'Result<Foo, Err>' here. (1)"));
-		};
+		let syn::Type::Path(ref mut type_path) = ty else { return self.untyped_client() };
```

Comment:
I don't really understand the effects of this return self.untyped_client() change; it looks like it turns some compile errors into successful returns; is this to play it safe because if the type the user returned isn't valid, we'll hit a type error anyways at this point?

Reply:
This is a really opinionated change, basically any type that isn't `Result<T, Error>`, `RpcResult<T>` or `ResponsePayload<T>` then I changed the client to generate `Result<serde_json::Value, Error>`.

To fix  https://github.com/paritytech/jsonrpsee/issues/1067

Reply:
Revert this, to have backward compatible stuff but technically right now the IntoResponse is breaking because it returns `jsonrpsee::core::server::ResponsePayload` instead of ` `jsonrpsee::types::ResponsePayload`